benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncClient.java

```diff
@@ -198,10 +199,16 @@ public void onCompleted() {
         histogram.recordValue((now - lastCall) / 1000);
         lastCall = now;
 
-        if (endTime > now) {
-          stub.unaryCall(request, this);
-        } else {
-          future.done();
+        Context prevCtx = null;
+        try {
+          prevCtx = Context.ROOT.attach();
```

Comment:
This line can be moved before the try, which will also prevent prevCtx from being `null` and throwing an exception in `detach()`